WithCustomDirectives is broken

You never initialize the `p.opts.customDirectives` map, leading to a `panic: assignment to entry in nil map` on line 95 of parser/parser.go.

Sure thing. Here's a minimum example demonstrating.

```go
package main

import (
	`log`

	"github.com/tufanbarisyildirim/gonginx/parser"
)

func main() {

	var err error
	var ngxParser *parser.Parser

	if ngxParser, err = parser.NewParser(
		"/etc/nginx/nginx.conf",
		parser.WithDefaultOptions(),
		parser.WithIncludeParsing(),
		parser.WithCustomDirectives(
			"foo",
			"foo_bar",
			"foo_bar_baz",
		),
		parser.WithSkipValidDirectivesErr(),
	); err != nil {
		log.Panicln(err)
	}

	_ = ngxParser
}
```

Here's the version I have imported for it:

```
$ grep -F gonginx go.mod 
require github.com/tufanbarisyildirim/gonginx v0.0.0-20241013191809-e73b7dd454e8
```
(which should be current default `HEAD`)

And here's the panic:

```
$ go run main.go 
panic: assignment to entry in nil map

goroutine 1 [running]:
main.main.WithCustomDirectives.func3(0xc0000ce410)
	/opt/dev/go/pkg/mod/github.com/tufanbarisyildirim/gonginx@v0.0.0-20241013191809-e73b7dd454e8/parser/parser.go:95 +0x70
github.com/tufanbarisyildirim/gonginx/parser.NewParserFromLexer(0xc0000b4240, {0xc0000caec0, 0x4, 0xc000000000?})
	/opt/dev/go/pkg/mod/github.com/tufanbarisyildirim/gonginx@v0.0.0-20241013191809-e73b7dd454e8/parser/parser.go:145 +0x162
github.com/tufanbarisyildirim/gonginx/parser.NewParser({0x4cc8e3, 0x15}, {0xc0000caec0, 0x4, 0x4})
	/opt/dev/go/pkg/mod/github.com/tufanbarisyildirim/gonginx@v0.0.0-20241013191809-e73b7dd454e8/parser/parser.go:129 +0x2ce
main.main()
	/tmp/testgo/main.go:14 +0xf4
exit status 2
```

--- a/parser/parser.go
+++ b/parser/parser.go
@@ -24,6 +24,17 @@ type options struct {
 	skipValidDirectivesErr     bool
 }
 
+func defaultOptions() options {
+	return options{
+		parseInclude:               false,
+		skipIncludeParsingErr:      false,
+		skipComments:               false,
+		customDirectives:           map[string]string{},
+		skipValidSubDirectiveBlock: map[string]struct{}{},
+		skipValidDirectivesErr:     false,
+	}
+}
+
 // Parser is an nginx config parser
 type Parser struct {
 	opts              options
@@ -70,7 +81,7 @@ func WithSkipIncludeParsingErr() Option {
 // WithDefaultOptions default options
 func WithDefaultOptions() Option {
 	return func(p *Parser) {
-		p.opts = options{}
+		p.opts = defaultOptions()
 	}
 }
 
@@ -136,7 +147,7 @@ func NewParserFromLexer(lexer *lexer, opts ...Option) *Parser {
 	configRoot, _ := filepath.Split(lexer.file)
 	parser := &Parser{
 		lexer:          lexer,
-		opts:           options{customDirectives: make(map[string]string), skipValidSubDirectiveBlock: map[string]struct{}{}},
+		opts:           defaultOptions(),
 		parsedIncludes: make(map[*config.Include]*config.Config),
 		configRoot:     configRoot,
 	}

